src/VS.Mefx/ConfigCreator.cs

```diff
@@ -0,0 +1,491 @@
+            if (this.Options.Folders != null)
+            {
+                IEnumerable<string> folders = this.Options.Folders;
+                foreach (string folder in folders)
+                {
+                    string folderPath = Path.GetFullPath(Path.Combine(currentFolder, folder));
+                    if (Directory.Exists(folderPath))
+                    {
+                        this.SearchFolder(folderPath);
+                    }
+                    else
+                    {
+                        string missingFolderMessage = string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.MissingFolderMessage,
+                            folder);
+                        this.Options.ErrorWriter.WriteLine(missingFolderMessage);
+                    }
+                }
+            }
+
+            this.OutputCacheFile = options.CacheFile;
+        }
+
+        /// <summary>
+        /// Gets or sets the command line arguments specified by the user.
+        /// </summary>
+        private CLIOptions Options { get; set; }
+
+        /// <summary>
+        /// Gets the catalog that stores information about the imported parts.
+        /// </summary>
+        public ComposableCatalog? Catalog { get; private set; }
+
+        /// <summary>
+        /// Gets configuration information associated with the imported parts.
+        /// </summary>
+        public CompositionConfiguration? Config { get; private set; }
+
+        /// <summary>
+        /// Gets or sets a dictionary storing parts indexed by thier parts name for easy lookup.
+        /// </summary>
+        public Dictionary<string, ComposablePartDefinition> PartInformation { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path of the cache file to store the processed parts.
+        /// </summary>
+        private string? OutputCacheFile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the paths to the assembly files we want to read.
+        /// </summary>
+        private List<string> AssemblyPaths { get; set; }
+
+        /// <summary>
+        /// Gets or sets the paths to the cache files we want to read.
+        /// </summary>
+        private List<string> CachePaths { get; set; }
+
+        /// <summary>
+        /// Method to get the details about a part, i.e. the part Definition, given its name.
+        /// </summary>
+        /// <param name="partName"> The name of the part we want to get details about.</param>
+        /// <returns><see cref="ComposablePartDefinition"/> associated with the given part if it is
+        /// present in the catalog and null otherwise.</returns>
+        public ComposablePartDefinition? GetPart(string partName)
+        {
+            if (!this.PartInformation.ContainsKey(partName))
+            {
+                return null;
+            }
+
+            return this.PartInformation[partName];
```

Comment:
This results in two dictionary lookups. You can optimize this to just one by using this syntax:

```cs
this.PartInformation.TryGetValue(partName, out ComposablePartDefinition? partDefinition);
return partDefinition;
```